legend-engine-shared-core/src/main/java/org/finos/legend/engine/shared/core/identity/CredentialVisitor.java

```diff
@@ -12,14 +12,22 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package org.finos.legend.engine.plan.execution.stores.service.plugin;
+package org.finos.legend.engine.shared.core.identity;
 
-import org.finos.legend.engine.plan.execution.stores.StoreExecutor;
+import org.finos.legend.engine.shared.core.identity.credential.ApiTokenCredential;
+import org.finos.legend.engine.shared.core.identity.credential.PlaintextUserPasswordCredential;
 
-public class ServiceStore
+public interface CredentialVisitor<T>
 {
-    public static StoreExecutor build()
+    default T visit(PlaintextUserPasswordCredential credential)
```

Comment:
I understand why you have this. But this doesn't scale well when we try to introduce new credential types as we can't have those in core and can't write such signature which essentially lead us mid-way 
Maybe we can just have 1 function/api
`T visit(Credential credential)`

what do you think?